Bm/24 profile customization

Ben-Mast opened the pull request:
As a user, I want to customize my account and profile so that it reflects my personal information and preferences.

Acceptance Criteria:

🛑User can upload a profile picture (you can upload, but the backend doesn't do anything with it, because we need a good way to store and retrieve the images. Best way to do that would probably be a cloud server like firebase, too lazy to set that up rn)
✅User can update personal information (name, age, fitness level....)
✅User can set privacy preferences for their profile (doesn't do anything right now until we implement social stuff)
✅Changes are saved and displayed correctly after update

We'll probably change the way the user relation is set up, but I kept it simple for now. The user can go into settings and change all of their personal information, along with their username and password. They can also change some profile settings, like their display name and privacy settings. Most of the inputs have verification with messages to the user telling them when they try to submit invalid inputs.

ranegray commented on ProjectSourceCode/index.js:
```diff
@@ -167,17 +173,169 @@ app.get('/activity', auth, async (req, res) => {
+    }
+})
+
+app.post('/settings/profile', upload.single('profilePicture'), async (req, res) => {
```
Disregard my previous comment about dynamic api routes. We can keep these separated but I do think that this one could use a little love similar to the changes I recommended above.

ranegray reviewed (changes requested):
Added a few comments, feel free to completely ignore them and just send it lol let me know what you think.

Ben-Mast commented:
> Added a few comments, feel free to completely ignore them and just send it lol let me know what you think.

I tried to implement all of the changes. The post routes look a lot cleaner, and use the code that you suggested. I implemented frontend file limit size for the profile photo, still haven't done anything with it on the backend yet though. I split up the nav bar and sub nav-bar logic. I added green and red messages depending on error or success. I also fixed the username bug on the activity page. Let me know what you think!

ranegray reviewed (approved):
Noice!
